.github/workflows/build.yaml

```diff
@@ -64,24 +94,50 @@ jobs:
               repo: context.repo.repo,
               per_page: 500,
               workflow_id,
-              status: 'success',,
             });
             for (const run of runs.workflow_runs) {
-              if (head_sha === run.head_sha) return run.html_url
-              else if (tree_id === run.head_commit?.tree_id) return run.html_url
+              let reason
+              if (head_sha === run.head_sha) reason = 'the same commit'
+              else if (tree_id === run.head_commit?.tree_id) reason = 'a tree-same commit'
+              else continue
+
+              const { data: jobs } = await github.rest.actions.listJobsForWorkflowRun({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                run_id: run.id
+              })
+              for (const job of jobs.jobs) {
+                if (job.conclusion !== 'success') continue
+                const jobLabel = job.name.startsWith('Build') ? 'build' : 'functional_test'
```

Comment:
Matching by job name feels fragile.. is there another, more robust way?

Reply:
Sadly, no. GitHub's REST API reports the job's labels as they are _displayed_, not as they are _defined_.